spec/models/user_spec.rb

```diff
@@ -510,6 +510,70 @@
     end
   end
 
+  describe '#personal_key_generated_at' do
+    let(:user) do
+      build(:user, encrypted_recovery_code_digest_generated_at: digest_generated_at)
+    end
+    let(:digest_generated_at) { nil }
+
+    context 'the user has a encrypted_recovery_code_digest_generated_at date' do
+      let(:digest_generated_at) { 1.day.ago }
+
+      it 'returns the date in the digest' do
+        expect(
+          user.personal_key_generated_at,
+        ).to be_within(1.second).of(digest_generated_at)
```

Comment:
I find `be_within` to be a code smell, since it signals possible race conditions based on timing that might lead to flakey test failures. In this case, it also doesn't seem necessary, since we should expect it to be strictly equal to the time we assigned when building the user?

```suggestion
        expect(user.personal_key_generated_at).to eq(digest_generated_at)
```

(Same note applies to other use of `be_within` in these added specs)